Editor/SettingsAnalysis/UniversalRenderPipelineAnalyzer.cs

```diff
@@ -0,0 +1,182 @@
+            {
+                yield return ProjectIssue.Create(IssueCategory.ProjectSetting, k_URPAssetDescriptor)
+                    .WithLocation("Project/Graphics");
+            }
+            else
+            {
+#if UNITY_ANDROID || UNITY_IOS || UNITY_SWITCH
+                foreach (var analyzeSrpAsset in RenderPipelineUtils.AnalyzeAssets(Analyze)) yield return analyzeSrpAsset;
+
+                var allCameraData = RenderPipelineUtils
+                    .GetAllComponents<UniversalAdditionalCameraData>();
+                foreach (var cameraData in allCameraData)
+                {
+                    if (cameraData.stopNaN)
+                        yield return ProjectIssue.Create(IssueCategory.ProjectSetting,
+                            k_CameraStopNanDescriptor);
+                }
+#endif
+            }
+#else
+            yield break;
+#endif
+        }
+
+        private static void FixHdrSetting(ProjectIssue issue)
+        {
+#if UNITY_2019_3_OR_NEWER
+            RenderPipelineUtils.FixAssetSetting(issue, p => SetHdrSetting(p, false));
+#endif
+        }
+
+        private static void FixMsaaSampleCountSetting(ProjectIssue issue)
+        {
+#if UNITY_2019_3_OR_NEWER
+            RenderPipelineUtils.FixAssetSetting(issue, p => SetMsaaSampleCountSetting(p, 2));
+#endif
+        }
+
+#if UNITY_2019_3_OR_NEWER
+        private IEnumerable<ProjectIssue> Analyze(RenderPipelineAsset renderPipeline, int qualityLevel)
+        {
+#if PACKAGE_URP
+            bool? supportsHDR = GetHdrSetting(renderPipeline);
+            if (supportsHDR != null && supportsHDR.Value)
+            {
+                yield return RenderPipelineUtils.CreateAssetSettingIssue(qualityLevel, renderPipeline.name,
+                    k_HdrSettingDescriptor);
+            }
+
+            int? msaaSampleCount = GetMsaaSampleCountSetting(renderPipeline);
+            if (msaaSampleCount != null && msaaSampleCount >= 4)
+            {
+                yield return RenderPipelineUtils.CreateAssetSettingIssue(qualityLevel, renderPipeline.name,
+                    k_MsaaSampleCountSettingDescriptor);
+            }
+#else
+            yield break;
+#endif
+        }
+
+        internal static bool? GetHdrSetting(RenderPipelineAsset renderPipeline)
+        {
+            if (renderPipeline == null) return null;
+#if PACKAGE_URP
+            if (renderPipeline is UniversalRenderPipelineAsset urpAsset)
+            {
+                return urpAsset.supportsHDR;
+            }
+#endif
+            return null;
+        }
+
+        internal static void SetHdrSetting(RenderPipelineAsset renderPipeline, bool value)
+        {
+            if (renderPipeline == null) return;
+#if PACKAGE_URP
+            if (renderPipeline is UniversalRenderPipelineAsset urpAsset)
+            {
+                urpAsset.supportsHDR = value;
+            }
+#endif
+        }
+
+        internal static int? GetMsaaSampleCountSetting(RenderPipelineAsset renderPipeline)
```

Comment:
Again, what is the advantage of a nullable int?

Reply:
Same as above, but I can just return -1 in case of RP is null.